components/collection-table.js

```diff
@@ -0,0 +1,546 @@
+// node_modules
+import { motion } from "framer-motion"
+import {
+  CheckIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ClipboardCopyIcon,
+} from "@heroicons/react/solid"
+import { useRouter } from "next/router"
+import PropTypes from "prop-types"
+import { useContext, useEffect, useMemo, useRef, useState } from "react"
+// components
+import Button from "./button"
+import Checkbox from "./checkbox"
+import CopyButton from "./copy-button"
+import { DataGridContainer } from "./data-grid"
+import GlobalContext from "./global-context"
+import Icon from "./icon"
+import Instruction from "./instruction"
+import Modal from "./modal"
+import Paragraph from "./paragraph"
+import SortableGrid from "./sortable-grid"
+// libs
+import {
+  clearHiddenColumnsFromUrl,
+  extractHiddenColumnIdsFromUrl,
+  filterHiddenColumns,
+  flattenCollection,
+  generateHiddenColumnsUrl,
+  generateTableColumns,
+  loadStoredHiddenColumns,
+  saveStoredHiddenColumns,
+  sortColumns,
+} from "../libs/collection-table"
+
+/**
+ * Displays the buttons to hide or show all columns at once.
+ */
+const ChangeAllControls = ({ onChangeAllHiddenColumns }) => {
+  const className = "flex-grow md:flex-grow-0"
+  return (
+    <div className="flex gap-1">
+      <Button
+        className={className}
+        onClick={() => onChangeAllHiddenColumns(false)}
+      >
+        Show All Columns
```

Comment:
We’ll need to talk about that as a group as soon as possible. I saw enough people who accidentally used title case in encoded (“Project Overview” in the help menu; many others were corrected) that I felt just using title case for title-like things would cause less friction. When I started here, I thought using sentence case for title-like things was weird, and it took me a while to stop accidentally using title case.